go/consensus/tendermint/abci/mux.go

```diff
@@ -995,7 +996,7 @@ func (mux *abciMux) ApplySnapshotChunk(req types.RequestApplySnapshotChunk) type
 
 	// Check if we are done with the restoration. In this case, finalize the root.
 	if done {
-		err = mux.state.storage.NodeDB().Finalize(mux.state.ctx, cp.Root.Version, []hash.Hash{cp.Root.Hash})
+		err = mux.state.storage.NodeDB().Finalize(mux.state.ctx, cp.Root.Version, []storageApi.Root{cp.Root})
```

Comment:
In case you specify roots, why would you need to also specify the version as a separate argument? Finalize should just make sure that all the passed roots use exactly the same version.